### PR TITLE
[Bugfix] Prevent Navigation Modal Fixes

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes slight changes to avoid having debounce when loading pages and sets it to a very small value that users cannot even perceive. Let me know your thoughts.

### CODE REVIEW

**Comment by Civolilah** [comment]
@turbo124 There is only one edge case that should be covered: uploading documents to the invoice triggers query invalidation for invoices, but uploading documents does NOT update the `updated_at` property in the invoice object. This unnecessarily triggers the modal after uploading documents.

Would it be too much work for you to update the `updated_at` property on the invoice once the docs are uploaded/deleted/updated so we can avoid this case?

File: src/components/PreventNavigationModal.tsx [hunk 1]
```diff
@@ -114,6 +117,8 @@ export function PreventNavigationModal() {
           {t('error_unsaved_changes')}
         </span>
 
+        {changes && <span>{JSON.stringify(changes)}</span>}
```
[the remark this answers is not in the document]
Comment (beganovich) [reply src/components/PreventNavigationModal.tsx:120]: Can you please put this behind the build flag (env variable)?

Make sure it's false by default and ping Dave to enable it so we can get info while testing. Thanks.

[the remark this answers is not in the document]
Comment (Civolilah) [reply src/components/PreventNavigationModal.tsx]: @beganovich Sure thing, it has been implemented and is controlled by the `VITE_ENABLE_DISCARD_CHANGES_TRACKING` env variable.

**Comment by Civolilah** [comment]
@turbo124 I just updated the PR with a setting build flag for displaying a stringified object in the discard changes modal, which indicates the reason for displaying the modal. To enable this part of the functionality, simply set the `VITE_ENABLE_DISCARD_CHANGES_TRACKING` env variable to `true`.

**Review by beganovich — APPROVED** [review]

**Pull request merged**